colossalai/zero/init_ctx/init_context.py

```diff
@@ -82,6 +83,31 @@ def _post_context_exec(self):
         pass
 
 
+class ZeroContextConfig(object):
+
+    def __init__(self,
+                 previous_config=None,
+                 convert_fp16: bool = False,
+                 shard_param: bool = False,
+                 rm_torch_payload_on_the_fly: bool = False):
+        super().__init__()
+        self.prev_config: Optional[ZeroContextConfig] = previous_config
+        self.convert_fp16: bool = convert_fp16
+        self.shard_param: bool = shard_param
+        self.rm_torch_payload_on_the_fly: bool = rm_torch_payload_on_the_fly
+
+
+ZERO_CONTEXT = ZeroContextConfig()
+
+
+@contextlib.contextmanager
+def modify_zero_context_config(**kwargs):
+    global ZERO_CONTEXT
+    ZERO_CONTEXT = ZeroContextConfig(ZERO_CONTEXT, **kwargs)
```

Comment:
https://www.codefactor.io/repository/github/hpcaitech/colossalai/pull/546
can you avoid to use global variable?

Reply:
done